### ISSUE TITLE
"replenishing optimization" works different from the documentation

### ISSUE THREAD

**Issue by hojongs** [reactor/reactor-core#1989]
I read "replenishing optimization" part on Reactor Reference documentation.
then, I wrote some code and executed it.

and I think it's different from that part of the documentation

the part of the documentation is here:
> Prefetch is a way to tune the initial request made on these inner sequences. If unspecified, most of these operators start with a demand of 32.
> 
> These operators usually also implement a replenishing optimization: Once the operator has seen 25% of the prefetch request fulfilled, it re-requests 25% from upstream. This is a heuristic optimization made so that these operators proactively anticipate the upcoming requests.

https://projectreactor.io/docs/core/release/reference/#_operators_that_change_the_demand_from_downstream

and the code is here:
```java
Flux.range(1, 200)
    .log()
    .publishOn(Schedulers.single(), 100)
    .subscribe()
```

## Expected Behavior

According to the documentation, 
1. Prefetch 100 items from publishOn() to range()
2. Push 25% from upstream (= from range())
3. Re-request 25% to upstream (= from publishOn() to range())
4. Repeat 2~3 process above until done

## Actual Behavior

1. Prefetch 100 items from publishOn() to range()
2. Push **75%** from upstream (= from range())
3. Re-request **75%** to upstream (= from publishOn() to range())
4. Repeat 2~3 process above until done

## Steps to Reproduce

```java
@Test
public void repoCase() {
    Flux.range(1, 200)
        .log()
        .publishOn(Schedulers.single(), 100)
        .subscribe()
}
```

```
[ INFO] (Test worker) | onSubscribe([Synchronous Fuseable] FluxRange.RangeSubscription)
[ INFO] (Test worker) | request(100)
[ INFO] (Test worker) | onNext(1)
[ INFO] (Test worker) | onNext(2)
[ INFO] (Test worker) | onNext(3)
[ INFO] (Test worker) | onNext(4)
[ INFO] (Test worker) | onNext(5)
[ INFO] (Test worker) | onNext(6)
[ INFO] (Test worker) | onNext(7)
[ INFO] (Test worker) | onNext(8)
[ INFO] (Test worker) | onNext(9)
[ INFO] (Test worker) | onNext(10)
[ INFO] (Test worker) | onNext(11)
[ INFO] (Test worker) | onNext(12)
[ INFO] (Test worker) | onNext(13)
[ INFO] (Test worker) | onNext(14)
[ INFO] (Test worker) | onNext(15)
[ INFO] (Test worker) | onNext(16)
[ INFO] (Test worker) | onNext(17)
[ INFO] (Test worker) | onNext(18)
[ INFO] (Test worker) | onNext(19)
[ INFO] (Test worker) | onNext(20)
[ INFO] (Test worker) | onNext(21)
[ INFO] (Test worker) | onNext(22)
[ INFO] (Test worker) | onNext(23)
[ INFO] (Test worker) | onNext(24)
[ INFO] (Test worker) | onNext(25)
[ INFO] (Test worker) | onNext(26)
[ INFO] (Test worker) | onNext(27)
[ INFO] (Test worker) | onNext(28)
[ INFO] (Test worker) | onNext(29)
[ INFO] (Test worker) | onNext(30)
[ INFO] (Test worker) | onNext(31)
[ INFO] (Test worker) | onNext(32)
[ INFO] (Test worker) | onNext(33)
[ INFO] (Test worker) | onNext(34)
[ INFO] (Test worker) | onNext(35)
[ INFO] (Test worker) | onNext(36)
[ INFO] (Test worker) | onNext(37)
[ INFO] (Test worker) | onNext(38)
[ INFO] (Test worker) | onNext(39)
[ INFO] (Test worker) | onNext(40)
[ INFO] (Test worker) | onNext(41)
[ INFO] (Test worker) | onNext(42)
[ INFO] (Test worker) | onNext(43)
[ INFO] (Test worker) | onNext(44)
[ INFO] (Test worker) | onNext(45)
[ INFO] (Test worker) | onNext(46)
[ INFO] (Test worker) | onNext(47)
[ INFO] (Test worker) | onNext(48)
[ INFO] (Test worker) | onNext(49)
[ INFO] (Test worker) | onNext(50)
[ INFO] (Test worker) | onNext(51)
[ INFO] (Test worker) | onNext(52)
[ INFO] (Test worker) | onNext(53)
[ INFO] (Test worker) | onNext(54)
[ INFO] (Test worker) | onNext(55)
[ INFO] (Test worker) | onNext(56)
[ INFO] (Test worker) | onNext(57)
[ INFO] (Test worker) | onNext(58)
[ INFO] (Test worker) | onNext(59)
[ INFO] (Test worker) | onNext(60)
[ INFO] (Test worker) | onNext(61)
[ INFO] (Test worker) | onNext(62)
[ INFO] (Test worker) | onNext(63)
[ INFO] (Test worker) | onNext(64)
[ INFO] (Test worker) | onNext(65)
[ INFO] (Test worker) | onNext(66)
[ INFO] (Test worker) | onNext(67)
[ INFO] (Test worker) | onNext(68)
[ INFO] (Test worker) | onNext(69)
[ INFO] (Test worker) | onNext(70)
[ INFO] (Test worker) | onNext(71)
[ INFO] (Test worker) | onNext(72)
[ INFO] (Test worker) | onNext(73)
[ INFO] (Test worker) | onNext(74)
[ INFO] (Test worker) | onNext(75)
[ INFO] (Test worker) | onNext(76)
[ INFO] (single-1) | request(75)
[ INFO] (Test worker) | onNext(77)
[ INFO] (Test worker) | onNext(78)
[ INFO] (Test worker) | onNext(79)
...
```

## Your Environment

* Reactor version(s) used: 3.3.1.RELEASE
* Other relevant libraries versions (eg. `netty`, ...):
* JVM version (`javar -version`): JDK 8
* OS and version (eg `uname -a`):

**Comment by simonbasle** [comment]
Ah yes there is indeed a typo: `25%` should be `75%`, good catch!
Would you like to fix that in a PR? (if so, please do it on `3.2.x` branch)

**Comment by hojongs** [comment]
Thank you for reply.
Ok, I'll do it :)
